### PR TITLE
Add query block patterns

### LINKED ISSUE #13: Block Patterns: Add Query Patterns
Query Patterns

These mockups are available with more detail in [the theme's Figma file](https://www.figma.com/file/76mfUcaK4QDlrXElk8MK3H/?node-id=758%3A4770).

### PR DESCRIPTION
**Pull request by kjellr**
Closes #13. 

⚠️ _This one relies on some of the block patterns setup from #74, so the PR builds on that for now._

---

This PR adds the 5 query patterns from [the original mockups](https://www.figma.com/file/76mfUcaK4QDlrXElk8MK3H/Twenty-Twenty-Two?node-id=758%3A4770). 

---

**Default Posts**

default

This one mirrors the default index template. 

Known issues: 

- There's no current way to enforce a responsive square shape for featured images. (https://github.com/WordPress/gutenberg/issues/35524)

---

**Text-based grid of posts**

Screen Shot 2021-10-11 at 3 03 34 PM

---

**Simple Blog Posts**

Screen Shot 2021-10-11 at 3 11 06 PM

Known issues: 

- The margin set on the post titles does not appear correct on the front-end. (https://github.com/WordPress/gutenberg/issues/35522)

---

**Image grid posts**

image-grid

Known issues:

- Post date is supposed to be stacked. This would be solved by the addition of a flex-wrap option. (https://github.com/WordPress/gutenberg/issues/35525)
- Post date can not be formatted into the "11.11.2021" format as originally designed. (https://github.com/WordPress/gutenberg/issues/34762)

---

**Large post titles**

large-titles

Known issues: 

- - Post date can not be formatted into the "Nov 11" format as originally designed. (https://github.com/WordPress/gutenberg/issues/34762)

---

**Irregular posts grid**

irregular-grid

Known issues: 

- Since this requires use of multiple query blocks, it cannot include post pagination.

### CODE REVIEW

File: inc/patterns/query-default.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Default posts block pattern
+ */
+return array(
+	'title'      => __( 'Default posts', 'twentytwentytwo' ),
+	'categories' => array( 'twentytwentytwo-query' ),
+	'blockTypes' => array( 'core/query' ),
+	'content'    => '<!-- wp:query {"query":{"offset":0,"postType":"post","categoryIds":[],"tagIds":[],"order":"desc","orderBy":"date","author":"","search":"","sticky":""},"displayLayout":{"type":"flex","columns":3},"align":"wide","layout":{"inherit":true}} -->
```
Comment (jffng) [inline inc/patterns/query-default.php]: Should this have a `"perPage": 6` (or `9`) attribute as well? It defaults to 3.

File: inc/patterns/query-image-grid.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Image grid posts block pattern
+ */
+return array(
+	'title'      => __( 'Image grid posts', 'twentytwentytwo' ),
+	'categories' => array( 'twentytwentytwo-query' ),
+	'blockTypes' => array( 'core/query' ),
+	'content'    => '<!-- wp:query {"query":{"offset":0,"postType":"post","categoryIds":[],"tagIds":[],"order":"desc","orderBy":"date","author":"","search":"","sticky":"","inherit":false},"displayLayout":{"type":"flex","columns":3},"layout":{"inherit":true}} -->
```
Comment (jffng) [inline inc/patterns/query-image-grid.php]: Same comment as above: should this have a `"perPage": 6` (or `9`) attribute as well?

File: inc/patterns/query-text-grid.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Text-based grid of posts block pattern
+ */
+return array(
+	'title'      => __( 'ext-based grid of posts', 'twentytwentytwo' ),
```
Comment (jffng) [inline inc/patterns/query-text-grid.php]:
```suggestion
	'title'      => __( 'Text-based grid of posts', 'twentytwentytwo' ),
```

File: inc/patterns/query-large-titles.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Large post titles block pattern
```
Comment (jffng) [inline inc/patterns/query-large-titles.php:3]: This one is looking a little funky in the editor vs front-end:

Editor | View
------- | -------
Screen Shot 2021-10-11 at 5 22 04 PM | Screen Shot 2021-10-11 at 5 21 30 PM

File: inc/patterns/query-text-grid.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	'categories' => array( 'twentytwentytwo-query' ),
+	'blockTypes' => array( 'core/query' ),
+	'content'    => '<!-- wp:query {"query":{"offset":0,"postType":"post","categoryIds":[],"tagIds":[],"order":"desc","orderBy":"date","author":"","search":"","sticky":""},"displayLayout":{"type":"flex","columns":3},"align":"wide","layout":{"inherit":true}} -->
```
Comment (jffng) [inline inc/patterns/query-text-grid.php]: Same comment re: `perPage`

**Review by jffng — COMMENTED** [review]
These are looking really good. I left some minor comments.

File: inc/patterns/query-large-titles.php (hunk 4 above)
Comment (kjellr) [reply inc/patterns/query-large-titles.php:3]: Yeah, I noticed that too -- I want to do a little digging and see what's causing it... I remember Gutenberg doing that a while ago but I thought it was fixed.

File: inc/patterns/query-default.php (hunk 1 above)
Comment (kjellr) [reply inc/patterns/query-default.php]: I'm not sure about the `perPage` values. There was [some recent discussion here](https://github.com/WordPress/gutenberg/issues/32638) about the number of query items per pattern. I thought maybe leading this out would inherit a decent default value, but 3 seems wrong. I'll add something more specific I suppose.

File: inc/patterns/query-large-titles.php (hunk 4 above)
Comment (kjellr) [reply inc/patterns/query-large-titles.php:3]: I filed an upstream issue here for this one: https://github.com/WordPress/gutenberg/issues/35551

**Comment by jffng** [comment]
Once #74 is in, can you switch the target branch to trunk to get the tests run? Assuming those are good, I say 🚢

**Comment by kjellr** [comment]
@jffng Seems like that worked. Mind giving it one more look?

**Review by jffng — APPROVED** [review]

**Pull request merged**